### DISCUSSION TITLE
Notifications

### DISCUSSION THREAD

**Discussion by msaulnier** [InfiniTimeOrg/InfiniLink#46] [General]
Hi guys, 

I have been beta testing the Infinilink 0.9. 3 on my IOS  phone for a week and I was thinking. I know that we can't get the app to send notification because we need to use the ANS but what if we could use the IOS short cut  automation to send the text to a variable and then in that automation open the infinilink app to then call the send Notification option in the app pasting the text message variable? 
I know this sound clunky but just a thought?  Besides I'm still trying to understand those IOS shortcuts and I haven't figured how to get the text message first anyway.

**Comment by xan-m** [comment]
Hello, and thanks for the suggestion! I have looked into adding Shortcuts functionality to the app. Honestly, it's an extremely convoluted and pretty poorly documented process to add it to your app, and it requires adding Siri functionality as well. I ran into a wall with it and ended up shifting my focus back to core features. Also, while I think it might be nice to create custom notifications using Shortcuts, I don't think it's an option for redirecting system notifications (e.g. texts, calls, emails, etc.). 

If you're looking to send arbitrary notifications to the phone, you can do so using the "send notification" button in the settings menu. You can use this functionality to send a short list or reminder to your watch.

With all that said, I will take another look at the Shortcuts stuff in the future. I think it would offer a nice bit of automation on top of standard notification support.

**Reply by msaulnier** [reply]
Thanks for answering. Yes IOS shortcuts is very poorly documented I could not agree more I have been trying to wrap my head around it for a couple of things and I still can't get past the basic samples they provide. 

I wasn't thinking this would be the option to solve for the notifications issue. But at the very least a clunky work around. :)  I was hoping that I might be able to set up a shortcut to call the "Send Notification" by calling that part of the app through an API call if that is possible.

Thanks I hope you find soemthing

One additional note I noticed that there is support for a Python IDE on ISO that can work with IOS Shortcuts I think I'llhave a look at that and there is a Javascript version too.  Might be an Idea?

**Comment by hellojaccc** [comment]
https://github.com/scandinave/InfiniTime

This is being worked on!

**Comment by liamcharger** [comment]
Notifications have been implemented in [#2217](https://github.com/InfiniTimeOrg/InfiniTime/pull/2217), so I'm closing this discussion now.
